AP-protocol.md

```diff
@@ -102,22 +96,29 @@ struct Query {
 	/// Unique identifier of the flood, to prevent loops.
 	flood_id: u64,
 	/// ID of client or server
-	initiator_id: RC<Arc<usize>>,
+	initiator_id: u64,
 	/// Time To Live, decremented at each hop to limit the query's lifespan.
+	/// When ttl reaches 0, we start a QueryResult message that reaches back to the initiator
 	ttl: u64,
 	/// Records the nodes that have been traversed (to track the connections).
-	path_trace: [u64; 20]
-	node_types: [NodeType; ]
+	path_trace: Vec<(u64, NodeType)>
 }
```

Comment:
```suggestion
}

enum NodeType{Client, Drone, Server}
```

Reply:
we could make NodeId reconizable like every drones id starts with D (D1 D2) every clients starts with C and server with C.

Reply:
For Node Id propose please refer to #19. In here anything is fine (it's just a temp Type).

Reply:
Done